src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,214 @@
+          smul_smul, inv_mul_cancel hc, one_smul],
+    have key : ∀ (u v : X), inner (f₁ u) (f₁ v) = inner u v := λ u v, by
+      rw [minor, function.comp_app, function.comp_app, real_inner_smul_left,
+          real_inner_smul_right, huv u v, ← mul_assoc, ← mul_assoc,
+          real.mul_self_sqrt $ le_of_lt $ inv_pos.mpr hc₁,
+          inv_mul_cancel $ ne_of_gt hc₁, one_mul],
+    exact ⟨f'.to_continuous_linear_map, h, c⁻¹, inv_ne_zero hc,
+           f₁.isometry_of_inner key, minor'⟩, },
+end
+
+/-- The characteristic function of a conformal map. -/
+def char_fun_at {f : X → Y} (x : X) {f' : X ≃L[ℝ] Y}
+(h : has_fderiv_at f f'.to_continuous_linear_map x) (H : conformal_at f x) : ℝ :=
+by choose c hc huv using (conformal_at_iff h).mp H; exact c
+
+/-- If a real differentiable map `f` is conformal at a point `x`,
+    then it preserves the angles at that point. -/
+theorem _root_.conformal_at_preserves_angle {f : X → Y} {x : X} {f' : X →L[ℝ] Y}
+(h : has_fderiv_at f f' x) (H : conformal_at f x) (u v : X) :
+  inner_product_geometry.angle (f' u) (f' v) = inner_product_geometry.angle u v :=
+begin
+  repeat { rw inner_product_geometry.angle },
+  suffices new : inner (f' u) (f' v) / (∥f' u∥ * ∥f' v∥) = inner u v / (∥u∥ * ∥v∥),
+  { rw new, },
+  { rcases H with ⟨f₁, h₁, c₁, hc₁, lie, h₂⟩,
+    have minor : ∥c₁∥ ≠ 0 := λ w, hc₁ (norm_eq_zero.mp w),
+    have : f' = f₁ := has_fderiv_at.unique h h₁,
+    rw [this, h₂],
+    repeat { rw function.comp_apply },
+    rw [real_inner_smul_left, real_inner_smul_right, ← mul_assoc, inner_map_map],
+    repeat { rw [norm_smul, norm_map] },
+    rw [← mul_assoc],
+    exact calc c₁ * c₁ * inner u v / (∥c₁∥ * ∥u∥ * ∥c₁∥ * ∥v∥)
+            = c₁ * c₁ * inner u v / (∥c₁∥ * ∥c₁∥ * ∥u∥ * ∥v∥) : by simp only [mul_comm, mul_assoc]
+        ... = c₁ * c₁ * inner u v / (abs c₁ * abs c₁ * ∥u∥ * ∥v∥) : by rw [real.norm_eq_abs]
+        ... = c₁ * c₁ * inner u v / (c₁ * c₁ * ∥u∥ * ∥v∥) : by rw [← pow_two, ← sq_abs, pow_two]
+        ... = c₁ * (c₁ * inner u v) / (c₁ * (c₁ * (∥u∥ * ∥v∥))) : by simp only [mul_assoc]
+        ... = (c₁ * inner u v) / (c₁ * (∥u∥ * ∥v∥)) : by rw mul_div_mul_left _ _ hc₁
+        ... = inner u v / (∥u∥ * ∥v∥) : by rw mul_div_mul_left _ _ hc₁, },
+end
+
+end conformal_at
+
+
+/-- A map `f` is conformal on a set `s` if it's conformal at every point in that set. -/
+def conformal_on {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+(f : X → Y) (s : set X) := ∀ (x : X), x ∈ s → conformal_at f x
+
+namespace conformal_on
+
+variables {X Y Z: Type*}
+  [inner_product_space ℝ X] [inner_product_space ℝ Y] [inner_product_space ℝ Z]
+
+theorem conformal_at {f : X → Y} {u : set X} (h : conformal_on f u) {x : X} (hx : x ∈ u) :
+conformal_at f x := h x hx
+
+theorem comp {f : X → Y} {g : Y → Z}
+{u : set X} {v : set Y} (hf : conformal_on f u) (hg : conformal_on g v) :
+conformal_on (g ∘ f) (u ∩ f⁻¹' v) := λ x hx, (hf x hx.1).comp (hg (f x) $ set.mem_preimage.mp hx.2)
```

Comment:
style: Again, following the "indent everything two spaces until the first `:=`" rule:
```suggestion
theorem comp {f : X → Y} {g : Y → Z}
  {u : set X} {v : set Y} (hf : conformal_on f u) (hg : conformal_on g v) :
  conformal_on (g ∘ f) (u ∩ f⁻¹' v) :=
λ x hx, (hf x hx.1).comp (hg (f x) $ set.mem_preimage.mp hx.2)
```

Reply:
Thank you for the edits! I suppose I need to indent in the lines after `theorem` and before `:=`?